src/nvim/ex_getln.c

```diff
@@ -4519,8 +4521,10 @@ add_to_history (
   histentry_T *hisptr;
   int len;
 
-  if (hislen == 0)              /* no history */
+  if (hislen == 0 || histype == HIST_INVALID) {  // no history
```

Comment:
Just in case, actually I do not see any code path which may lead to histype == HIST_INVALID.